### PR TITLE
Lightweight Storefront: Themes option in Settings

### LINKED ISSUE #11303: Feature flag

### PR DESCRIPTION
**Pull request by hafizrahman**
Closes: #11303, #11304

## Description

This PR:
- Adds feature flag for lightweight storefront,
- Adds a use case for checking lightweight storefront eligiblity,
- Adds a "Themes" setting in Settings > Store Settings area for eligible sites.

The "Themes" setting will not do anything yet, waiting for the themes selection flow to be built first.

## Testing instructions

1. Use a WordPress.com site (can be regular Atomic site or Woo Express)
2. Go to Settings, ensure "Themes" is listed in "Store Settings".

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift [hunk 1]
```diff
@@ -813,6 +824,11 @@ private extension SettingsViewController {
             comment: "Navigates to Install Jetpack screen."
         )
 
+        static let themes = NSLocalizedString(
+            "Themes",
+            comment: "Navigates to Themes screen."
+        )
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift]: Let's use the new style of localizing strings with keys and values.

```suggestion
        static let themes = NSLocalizedString(
            "settingsViewController.themesRow",
            value: "Themes",
            comment: "Navigates to Themes screen."
        )
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewModel.swift [hunk 2]
```diff
@@ -280,11 +280,17 @@ private extension SettingsViewModel {
         let storeSettingsSection: Section? = {
             var rows: [Row] = [.storeName]
 
-            let site = stores.sessionManager.defaultSite
-            if site?.isJetpackCPConnected == true ||
-                (site?.isNonJetpackSite == true &&
-                 featureFlagService.isFeatureFlagEnabled(.jetpackSetupWithApplicationPassword)) {
-                rows.append(.installJetpack)
+            if let site = stores.sessionManager.defaultSite {
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewModel.swift:283]: Good job cleaning up the optionals 💯

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewModel.swift [hunk 3]
```diff
@@ -280,11 +280,17 @@ private extension SettingsViewModel {
...
+                if site.isJetpackCPConnected == true ||
+                    (site.isNonJetpackSite == true &&
+                     featureFlagService.isFeatureFlagEnabled(.jetpackSetupWithApplicationPassword)) {
+                    rows.append(.installJetpack)
+                }
+
+                let themesUseCase = ThemeEligibilityUseCase(featureFlagService: featureFlagService)
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewModel.swift]: Nit: We can also keep the initializer empty as we default to the default feature flag service. Only explicit injection is needed for tests.

```suggestion
                let themesUseCase = ThemeEligibilityUseCase()
```

**Review by itsmeichigo — APPROVED** [review]
We should also add simple unit tests for `ThemeEligibilityUseCase` to improve our confidence. I'm approving this PR; please feel free to merge after adding them.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift (hunk 1 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift]: Thanks!

**Comment by hafizrahman** [comment]
@itsmeichigo thanks for the review! I've added the unit tests.

**Pull request merged**